cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/store/AppMetadataStore.java

```diff
@@ -343,6 +347,15 @@ public long getApplicationCount() throws IOException {
     return getApplicationSpecificationTable().count(ranges);
   }
 
+  public List<ApplicationMeta> getLatest(NamespaceId namespaceId, String appName) throws IOException {
+    // TODO: change this to filter by range of primary prefix key and index
+    return scanWithRange(
```

Comment:
@sanjanasandeep  I guess here we should do: scan by key range +`latest` index, return if it's not null; if the scan result is null, scan by with key range + DESC order of `created`, return the first element

Reply:
And before that, pls rebase onto`develop` and resolve conflicts if any

Reply:
yes, this is the TODO - but I was thinking of doing this in another PR like I mentioned before (since this involves adding LATEST as an index to the table and that involves modifying a few more tests) - making this change too large

Reply:
got it, let's try to merge this PR first